waltz-client/src/main/java/com/wepay/waltz/client/internal/network/WaltzNetworkClient.java

```diff
@@ -222,6 +232,45 @@ public void requestHighWaterMark(ReqId reqId) {
         }
     }
 
+    /**
+     * Requests list of partitions assigned on that server
+     *
+     * @return Completable future that will container the list of partitions assigned on that server
+     * @throws InterruptedException If thread interrupted while waiting for channel to be ready
+     */
+    public CompletableFuture<Object> getServerPartitionAssignments() throws InterruptedException {
```

Comment:
Why is this Future<Object> not Future<List<Integer>>?
Can you do casting using `future.thenApply()` so that we can have the right return type?

Reply:
Updated, thanks!